src/EventListener/PaymentLinkCreationListener.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Sylius\AdminOrderCreationPlugin\EventListener;
+
+use Doctrine\Common\Persistence\ObjectManager;
+use Payum\Core\Model\GatewayConfigInterface;
+use Payum\Core\Payum;
+use Payum\Core\Security\GenericTokenFactoryInterface;
+use Payum\Core\Security\TokenInterface;
+use Sylius\AdminOrderCreationPlugin\Sender\OrderPaymentLinkSenderInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\PaymentInterface;
+use Symfony\Component\EventDispatcher\GenericEvent;
+use Webmozart\Assert\Assert;
+
+final class PaymentLinkCreationListener
+{
+    /** @var Payum */
+    private $payum;
+
+    /** @var ObjectManager */
+    private $orderManager;
+
+    /** @var OrderPaymentLinkSenderInterface */
+    private $orderPaymentLinkSender;
+
+    /** @var string */
+    private $afterPayRoute;
+
+    public function __construct(
+        Payum $payum,
+        ObjectManager $orderManager,
+        OrderPaymentLinkSenderInterface $orderPaymentLinkSender,
+        string $afterPayRoute
+    ) {
+        $this->payum = $payum;
+        $this->orderManager = $orderManager;
+        $this->orderPaymentLinkSender = $orderPaymentLinkSender;
+        $this->afterPayRoute = $afterPayRoute;
+    }
+
+    public function setPaymentLink(GenericEvent $event): void
+    {
+        /** @var OrderInterface $order */
```

Comment:
```$order``` not needed in type hint section

Reply:
As far as I see it's required to get the autocomplete from PHPStorm